### DISCUSSION TITLE
jetstream registrations - email validation, dns check

### DISCUSSION THREAD

**Discussion by mindifest** [laravel/framework#34862] [Q&A] [answered]
Hi,
I am using Laravel 8.0 with jetstream 1.0 and wondering if there is a way to make email validation stronger, like add dns check..
Laravel makes it easy when're writing your own rules. I just have not found a way to hook into jetstream validation, if there is a way.

**Comment by devcircus** [comment, marked as answer]
Looks like, under the hood, Jetstream utilizes laravel/fortify. There, the source RegisteredUserController defers to the class "App\Actions\Fortify\CreateNewUser". You can see the file here [CreateNewUser.php](https://github.com/laravel/fortify/blob/1.x/stubs/CreateNewUser.php). After Jetstream is installed, that file will be stubbed out to "App\Actions\Fortify". The validation and creation logic is in that class and can be customized however you need.

If you need to change the registration routes or make other customizations that can't be made in that file, you can disable registration in the fortify config file and add your own custom route and controller.

**Reply by mindifest** [reply]
This is very helpful! Thank you!
